src/mlpack/methods/linear_regression/linear_regression_impl.hpp

```diff
@@ -164,11 +113,14 @@ typename LinearRegression<ModelMatType>::ElemType
 LinearRegression<ModelMatType>::Train(const MatType& predictors,
                                       const ResponsesType& responses,
                                       const WeightsType& weights,
-                                      const double lambda,
-                                      const bool intercept)
+                                      const std::optional<double> lambda,
+                                      const std::optional<bool> intercept)
 {
-  this->lambda = lambda;
-  this->intercept = intercept;
+  if (lambda.has_value())
+    this->lambda = lambda.value();
+  
+  if (intercept.has_value())
+    this->intercept = intercept.value();
```

Comment:
I dug into the test failure and actually the issue is on line 147/195 (I can't leave a comment there):

```
if (intercept)
```

Previously that referred to the boolean, but now it refers to the `std::optional`.  So either we would need to change the name of the input parameter, or change here to `if (this->intercept)`.  I don't think it matters either way, up to you.